plc4j/drivers/opcua/src/main/java/org/apache/plc4x/java/opcua/connection/OpcuaTcpPlcConnection.java

```diff
@@ -0,0 +1,469 @@
+        this( address, OPCUA_DEFAULT_TCP_PORT,  params, requestTimeout);
+        logger.info("Configured OpcuaTcpPlcConnection with: host-name {}", address.getHostAddress());
+    }
+
+    public OpcuaTcpPlcConnection(InetAddress address, int port, String params, int requestTimeout) {
+        this(params);
+        logger.info("Configured OpcuaTcpPlcConnection with: host-name {}", address.getHostAddress());
+        this.address = address;
+        this.port = port;
+        this.params = params;
+    }
+
+    public OpcuaTcpPlcConnection(String params) {
+        super(params);
+    }
+
+    public static OpcuaTcpPlcConnection of(InetAddress address, String params, int requestTimeout) {
+        return new OpcuaTcpPlcConnection(address, params, requestTimeout);
+    }
+
+    public static OpcuaTcpPlcConnection of(InetAddress address, int port, String params, int requestTimeout) {
+        return new OpcuaTcpPlcConnection(address, port, params, requestTimeout);
+    }
+
+
+
+    public InetAddress getRemoteAddress() {
+        return address;
+    }
+
+    @Override
+    public void connect() throws PlcConnectionException {
+        List<EndpointDescription> endpoints =  null;
+
+        try {
+            endpoints = DiscoveryClient.getEndpoints(getEndpointUrl(address, port, params)).get();
+        } catch (Exception ex) {
+            // try the explicit discovery endpoint as well
+            String discoveryUrl = getEndpointUrl(address, port, params);
+
+            if (!discoveryUrl.endsWith("/")) {
+                discoveryUrl += "/";
+            }
+            discoveryUrl += "discovery";
+
+            logger.info("Trying explicit discovery URL: {}", discoveryUrl);
+            try {
+                endpoints = DiscoveryClient.getEndpoints(discoveryUrl).get();
+            } catch (InterruptedException | ExecutionException e) {
+                throw new PlcConnectionException("Unable to discover URL:" + discoveryUrl);
+            }
+        }
+
+        EndpointDescription endpoint = endpoints.stream()
+                .filter(e -> e.getSecurityPolicyUri().equals(getSecurityPolicy().getUri()))
+                .filter(endpointFilter())
+                .findFirst()
+                .orElseThrow(() -> new PlcConnectionException("No desired endpoints from"));
+
+        OpcUaClientConfig config = OpcUaClientConfig.builder()
+            .setApplicationName(LocalizedText.english("eclipse milo opc-ua client"))
+            .setApplicationUri("urn:eclipse:milo:plc4x:client")
+            .setEndpoint(endpoint)
+            .setIdentityProvider(getIdentityProvider())
+            .setRequestTimeout(UInteger.valueOf(requestTimeout))
+            .build();
+
+        try {
+            this.client =  OpcUaClient.create(config);
+            this.client.connect().get();
+            isConnected = true;
+        } catch (UaException | InterruptedException | ExecutionException e) {
+            isConnected = false;
```

Comment:
We should log something in WARN, I think

Reply:
Same as below. So i will add a WARN first